### ISSUE TITLE
https_proxy and http_proxy environment variables are not respected

### ISSUE THREAD

**Issue by wolfoo2931** [pivotal-cf/go-pivnet#8]
We want to use this resource within a network which requires to a http proxy in order to connect to the internet. We configured the http_proxy and https_proxy in the concourse deployment manifest:

https://github.com/concourse/concourse/blob/master/jobs/groundcrew/spec#L124-L130

so that these environment variables show up in the container where the check/in/out scripts are executed. 

We noted that a go binary will respect these environment variables by default, except you provide transport configuration:

```
  	httpClient = &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{InsecureSkipVerify: c.skipSSLValidation},
		},
	}
```

In this case it seems to be necessary to pass the proxy configuration explicitly, like in this example: 

```
proxyUrl, err := url.Parse("http://proxyIp:proxyPort")
http.DefaultTransport = &http.Transport{Proxy: http.ProxyURL(proxyUrl)}
```

http://stackoverflow.com/a/14663620/6089918

**Comment by wolfoo2931** [comment]
We try to prepare a pull request if this change is welcome?

**Comment by robdimsdale** [comment]
@wolfoo2931 sure, a PR would be most welcome here! Let me know what you need to get started.

**Comment by wolfoo2931** [comment]
@robdimsdale We started to implement this. In fact this problem seems to be solved by adding this line

```
	httpClient = &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{InsecureSkipVerify: c.skipSSLValidation},
			Proxy: http.ProxyFromEnvironment,
		},
	}
```

into the MakeRequest function:
https://github.com/pivotal-cf/go-pivnet/blob/master/pivnet.go#L190-L194

But it turned out that testing this line is harder then expected because the environment variables are cached. This means it's not possible to set the http_proxy for one test and for the other not. We think it would be possible to extract the proxy tests into a separated file/testsuite. 

would this be accepted or is there another way which we don't see?

Best Regards,
Oliver

**Comment by robdimsdale** [comment]
Yes, that's challenging. You could test it with separate binaries as each binary execution can have its own environment overridden.

Depending on the complexity of the resulting PR, you could also not test this and it would probably be fine.

**Comment by robdimsdale** [comment]
Closed via #10
